src/lib/components/viewer/ViewerContext.svelte

```diff
@@ -189,7 +191,7 @@ layouts, stories, and tests.
     }
     $currentMode = mode;
     $currentNote = $currentDoc.notes?.find(noteMatchingPageHash) ?? null;
-    if (shouldPaginate(mode) && $currentPage !== hashPage) {
+    if (shouldPaginate(mode) && (hashPage || 0) > 1) {
```

Comment:
The comparison was made between $currentPage and hashPage because occasionally, especially at higher zoom levels, the viewer would jump from midway down the page you're looking at back up to the top of it. Can you explain how this solution works?

Reply:
I think this broke when I added the line to set `$currentPage` to fix the new `search` mode. So this fixes that. Can't promise we won't have to do another pass on pagination logic.